extensions/amp-story/1.0/amp-story-quiz.js

```diff
@@ -102,6 +137,31 @@ export class AmpStoryQuiz extends AMP.BaseElement {
     createShadowRootWithStyle(this.element, this.quizEl_, CSS);
   }
 
+  /** @override */
+  layoutCallback() {
+    return (this.responseDataPromise_ = this.element.hasAttribute('endpoint')
+      ? this.retrieveReactionData_()
+      : Promise.resolve());
+  }
+
+  /**
+   * Gets a Promise to return the unique AMP clientId
+   *
+   * @private
+   * @return {Promise<string>}
+   */
+  getClientId_() {
```

Comment:
LGTM. To confirm the id is an essential part of the feature and cannot be opt out?

Reply:
The ID is an essential part of the feature but there is fallback behavior for if the user has opted out of providing the `clientId`